packages/experimental/src/components/UserProfileImage/UserProfileImage.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright IBM Corp. 2020, 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Canary } from '../_Canary';
+import { pkg } from '../../settings';
+const componentName = 'UserProfileImage';
+
+import {
+  User16,
+  User20,
+  User24,
+  User32,
+  Group16,
+  Group20,
+  Group24,
+  Group32,
+} from '@carbon/icons-react';
+
+//import { expPrefix } from '../../global/js/settings';
+import cx from 'classnames';
+
+const blockClass = `${pkg.prefix}-user-profile-avatar`;
+
+export const UserProfileImage = !pkg.isComponentEnabled(componentName)
+  ? // Return canary if not released or flag not set
+    () => <Canary component={componentName} />
+  :({
+  backgroundColor,
+  icon = 'user',
+  initials,
+  image,
+  size = 'x-large',
+  theme
+}) => {
+
+  const icons = {
+    user: {
+      'x-small': <User16 />,
+      small: <User16 />,
+      medium: <User20 />,
+      large: <User24 />,
+      'x-large': <User32 />,
+    },
+    group: {
+      'x-small': <Group16 />,
+      small: <Group16 />,
+      medium: <Group20 />,
+      large: <Group24 />,
+      'x-large': <Group32 />,
+    },
+  };
+
+  const formatInitials = () => {
+    if (initials.length === 2) return initials;
+    return initials.match(/(^\S\S?|\b\S)?/g).join("").match(/(^\S|\S$)?/g).join("").toUpperCase()
+  };
+
+  const renderFillItem = () => {
+    if (image) {
+      return <img src={image} className={`${blockClass}-photo ${blockClass}-photo--${size}`} />;
+    } else {
+      if (initials) {
+        return formatInitials();
+      } else {
+        return icons[icon][size];
+      }
+    }
+  };
```

Comment:
This corresponds to the suggestion above to avoid instantiating ten render functions. Rather than have this be a function, we could make it a variable with a capital initial (so we can use it in JSX as a render function) and *return* a function based on the settings. That means that in the case of an image we create a function that returns the image tag, but in the case of initials we can just return the `formatInitials` function directly because it is a function that returns the required rendering (ie text), and in the case of needing an icon we can just return the appropriate icon from the structure directly because each of those too is a function that returns the required rendering.

```suggestion
      const FillItem = image
        ? () => (
            <img
              alt=""
              src={image}
              className={`${blockClass}-photo ${blockClass}-photo--${size}`}
            />
          )
        : initials
        ? formatInitials
        : icons[icon][size];
```